cfpb/sbl-frontend
# [e2e] Make tests runnable against dev

Made e2e test suite runnable against the dev deployment when running locally.

## Changes

- Added calls to cleanup logic after tests
- Added code to change the way auth works when running against dev
- Added additional environment variables that were missed before
- Fixed certificate issue causing the tests to run against dev

## How to test this PR

1. Read the notes
2. Make sure all other repos are up to date
3. Pull the branch for the sbl-frontend repo
4. Update your local .env file to reflect what is in the .env.example file
5. Comment out and override the following values in your local .env with what is listed below replacing the placeholder values with their appropriate ones:
    - SBL_PLAYWRIGHT_TEST_TARGET="<FRONTEND_URL>"
    - SBL_PLAYWRIGHT_TEST_REGTECH_TARGET="<USER_FI_URL>"
    - SBL_PLAYWRIGHT_TEST_FILING_TARGET="<FILING_URL>"
    - SBL_PLAYWRIGHT_TEST_MAIL_TARGET="<MAIL_URL>"
    - SBL_PLAYWRIGHT_TEST_CLEANUP_TARGET="<CLEANUP_URL>"
    - SBL_PLAYWRIGHT_TEST_KC_TARGET="<KEYCLOAK_URL>"
    - SBL_PLAYWRIGHT_TEST_KC_CLI_CLIENT_SECRET="<KEYCLOAK_SECRET>"

## Notes

- As long as dev is out of date and the tests have been updated then it is to be expected that you will see failures when running against dev.
- As it stands at the time of this writing, there are test failures related to dev being out of date.
